pkg/controller/kibanaassociation/association_controller.go

```diff
@@ -295,25 +286,30 @@ func (r *ReconcileAssociation) reconcileInternal(kibana kbtype.Kibana) (commonv1
 		return commonv1alpha1.AssociationPending, err
 	}
 
-	// update Kibana resource with ES access details
-	var expectedEsConfig kbtype.BackendElasticsearch
-	expectedEsConfig.CertificateAuthorities.SecretName = caSecretName
-	expectedEsConfig.URL = services.ExternalServiceURL(es)
-	expectedEsConfig.Auth.SecretKeyRef = association.ClearTextSecretKeySelector(&kibana, kibanaUserSuffix)
+	// construct the expected association configuration
+	authSecret := association.ClearTextSecretKeySelector(kibana, kibanaUserSuffix)
+	expectedESAssoc := &commonv1alpha1.AssociationConf{
+		AuthSecretName: authSecret.Name,
+		AuthSecretKey:  authSecret.Key,
+		CASecretName:   caSecretName,
+		URL:            services.ExternalServiceURL(es),
+	}
 
-	if !reflect.DeepEqual(kibana.Spec.Elasticsearch, expectedEsConfig) {
-		kibana.Spec.Elasticsearch = expectedEsConfig
+	// update the association configuration if necessary
+	if !reflect.DeepEqual(expectedESAssoc, kibana.AssociationConf()) {
 		log.Info("Updating Kibana spec with Elasticsearch backend configuration", "namespace", kibana.Namespace, "kibana_name", kibana.Name)
-		if err := r.Update(&kibana); err != nil {
+		if err := association.UpdateAssociationConf(r.Client, kibana, expectedESAssoc); err != nil {
+			log.Error(err, "Failed to update association configuration", "namespace", kibana.Namespace, "kibana_name", kibana.Name)
```

Comment:
IIRC we tried to not display the error when it's a `Conflict` because it's verbose and there's nothing the user or the operator can do but retry:

```
ERROR	kibana-association-controller	Failed to update association configuration	{"namespace": "default", "kibana_name": "kibana-sample", "error": "Operation cannot be fulfilled on kibanas.kibana.k8s.elastic.co \"kibana-sample\": the object has been modified; please apply your changes to the latest version and try again"}
github.com/elastic/cloud-on-k8s/vendor/github.com/go-logr/zapr.(*zapLogger).Error
	/Users/michael/go/src/github.com/elastic/cloud-on-k8s/vendor/github.com/go-logr/zapr/zapr.go:128
github.com/elastic/cloud-on-k8s/pkg/controller/kibanaassociation.(*ReconcileAssociation).reconcileInternal
	/Users/michael/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/kibanaassociation/association_controller.go:302
github.com/elastic/cloud-on-k8s/pkg/controller/kibanaassociation.(*ReconcileAssociation).Reconcile
	/Users/michael/go/src/github.com/elastic/cloud-on-k8s/pkg/controller/kibanaassociation/association_controller.go:162
github.com/elastic/cloud-on-k8s/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/Users/michael/go/src/github.com/elastic/cloud-on-k8s/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
github.com/elastic/cloud-on-k8s/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/Users/michael/go/src/github.com/elastic/cloud-on-k8s/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
github.com/elastic/cloud-on-k8s/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/Users/michael/go/src/github.com/elastic/cloud-on-k8s/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
github.com/elastic/cloud-on-k8s/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/Users/michael/go/src/github.com/elastic/cloud-on-k8s/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
github.com/elastic/cloud-on-k8s/vendor/k8s.io/apimachinery/pkg/util/wait.Until
	/Users/michael/go/src/github.com/elastic/cloud-on-k8s/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
```

https://github.com/elastic/cloud-on-k8s/blob/cca8fa1400da702f1b8910e1b110e6e507b362ce/pkg/controller/kibana/kibana_controller.go#L203-L207